libs/langchain/langchain/agents/agent.py

```diff
@@ -1263,3 +1266,74 @@ def _prepare_intermediate_steps(
             return self.trim_intermediate_steps(intermediate_steps)
         else:
             return intermediate_steps
+
+    def stream(
+        self,
+        input: Dict[str, Any],
+        config: Optional[RunnableConfig] = None,
+        **kwargs: Any,
+    ) -> Iterator[AgentStreamOutput]:
+        """Enables streaming over steps taken to reach final output."""
+        config = config or {}
+        iterator = AgentExecutorIterator(
+            self,
+            input,
+            config.get("callbacks"),
+            tags=config.get("tags"),
+            metadata=config.get("metadata"),
+            run_name=config.get("run_name"),
+            **kwargs,
+        )
+        for step in iterator:
+            if "intermediate_step" in step:
+                yield AgentStreamOutput(
+                    actions=[a for a, o in step["intermediate_step"]],
+                    observations=[o for a, o in step["intermediate_step"]],
+                )
+            else:
+                yield AgentStreamOutput(step)
+
+    async def astream(
+        self,
+        input: Dict[str, Any],
+        config: Optional[RunnableConfig] = None,
+        **kwargs: Any,
+    ) -> AsyncIterator[AgentStreamOutput]:
+        """Enables streaming over steps taken to reach final output."""
+        config = config or {}
+        iterator = AgentExecutorIterator(
+            self,
+            input,
+            config.get("callbacks"),
+            tags=config.get("tags"),
```

Comment:
kwargs also support tags/metadata/run_name, not sure if makes sense to raise a better error message saying that they have to be passed through config?

It would be good to document the other kwargs that are meant to be used